test/integration/restart/test_restart.py

```diff
@@ -70,3 +70,57 @@ def test_restart_with_auto_yes(tt_cmd, tmpdir_with_cfg):
 
     finally:
         app_cmd(tt_cmd, tmpdir_with_cfg, ["stop", app_name], [])
+
+
+def test_restart_no_args(tt_cmd, tmpdir):
+    test_app_path_src = os.path.join(os.path.dirname(__file__), "multi_app")
+
+    test_app_path = os.path.join(tmpdir, "multi_app")
+    shutil.copytree(test_app_path_src, test_app_path)
+
+    start_output = app_cmd(tt_cmd, test_app_path, ["start"], [])
+    assert "Starting an instance" in start_output[0]
+    wait_file(os.path.join(test_app_path, run_path, 'app1', 'master'), 'master.pid', [])
+    wait_file(os.path.join(test_app_path, run_path, 'app2'), 'app2.pid', [])
+    app1_prev_pid = app1_current_pid = 0
+    app2_prev_pid = app2_current_pid = 0
+
+    try:
+        with open(os.path.join(test_app_path, run_path, 'app1', 'master', 'master.pid')) \
+                as pid_file:
+            app1_prev_pid = int(pid_file.readline())
+
+        with open(os.path.join(test_app_path, run_path, 'app2', 'app2.pid')) as pid_file:
+            app2_prev_pid = int(pid_file.readline())
+
+        # Test confirmed restart.
+        restart_output = app_cmd(tt_cmd, test_app_path, ["restart"], ["y\n"])
+        assert "Confirm restart of all instances [y/n]" in restart_output[0]
+
+        wait_file(os.path.join(test_app_path, run_path, 'app1', 'master'), 'master.pid', [])
```

Comment:
Remove checking PIDs. This will lead to flaky tests. See #264 
Rely on output messages for now.

Reply:
Thank you, fixed.